src/python/WMCore/MicroService/DataStructs/MSOutputTemplate.py

```diff
@@ -231,11 +231,11 @@ def setRelVal(self, isRelVal):
             return True
         return False
 
-    def updateDoc(self, myDoc):
+    def updateDoc(self, myDoc, throw=False):
         """
-        Method to be used for updating the document fields from a dictionary
+        A method to be used for updating the document fields from a dictionary
         """
-        if self._checkAttr(self.docTemplate, myDoc, throw=False, update=False, **myDoc):
+        if self._checkAttr(self.docTemplate, myDoc, throw=throw, update=False, **myDoc):
```

Comment:
Can you please remind me why we would want NOT to throw an exception if the key is not in the template or not of the same data type?

Reply:
It is a method supposed to be used from outside the class, so since here we do not know how the user may want to handle those exceptions (if at all), I left the option for him/her to chose to either throw them or not. In general if it does, and the exception is not handled on the next step it will break the current run cycle. And there are cases where failing one and only one document is preferable than breaking the whole thread. (of course it got me into troubles when I tested it, because if not checked the exit status from this method, it may lead to silent errors  - but this is usually a responsibility of whom ever is about to call it).

Reply:
This method is actually a bit confusing. From the name, one can say nothing is supposed to get written, but just checked.
The "user" here seems to be the module itself, and there shouldn't be any other users. So I would actually implement what we need and what we will use, instead of providing flexibility that will never get used.

This can be addressed in a different PR though (setting it as unresolved).

Reply:
It was develpped in a quite early stage and the usage slightly changed afterwards. I would say if we start messing with that this may shake the ground for the whole Class, since it is constantly referred from any update method from the Class itself.